watcher/exts/playerstats.py

```diff
@@ -218,22 +267,50 @@ async def _equivalent_exchange(self, ctx, *, info):
         async with aiosqlite.connect(self.bot.db_path) as db:
             async with db.execute(f"select player_id, player_name, combined_stars, team_id, team_name, {rating} "
                                   f"from playerleagueandstars where league = '{other_league}' and "
-                                  f"(combined_stars > {combined_stars}-2 and combined_stars < {combined_stars}+2) "
-                                  f"group by player_id order by {rating} desc limit 10;") as cursor:
+                                  f"(combined_stars > {combined_stars}-2 and combined_stars < {combined_stars}+2) {on_team}"
+                                  f"group by player_id order by {rating} {sort_dir} {limit};") as cursor:
                 async for row in cursor:
                     other_players.append([row[0], row[1], row[2], row[3], row[4], row[5]])
+        team_caviat = f"on {on_team_name}" if on_team != '' else ''
         if len(other_players) < 1:
-            return await ctx.send(f"Could not find players within 2 stars of {player_name}.")
+            
+            return await ctx.send(f"Could not find players within 2 stars of {player_name} {team_caviat}")
         p_stars = round((combined_stars * 100)) / 100
-        response = f"Top 10 players by {raw_rating} within 2 combined stars of **{player_name}** " \
-                   f"({p_stars}) in {other_league} League\n\n"
-        for player in other_players[:10]:
-            o_player_name, team_name = player[1], player[4]
-            stars = round((player[2] * 100)) / 100
-            rating_star = player[5] * 5
-            rating = round((rating_star * 100)) / 100
-            response += f"**{o_player_name}**: {stars} ({team_name}) - {rating}⭐ {raw_rating}\n"
-        return await ctx.send(response)
+        if ouptut_format == 0:
+          if limit == '':
+            filter_desc = "All players"
+          elif sort_dir == 'asc':
+            filter_desc = f"Top {lim} players"
+          else:
+            filter_desc = f"Bottom {lim} players"
+
+
+          response = f"{filter_desc} by {raw_rating} within 2 combined stars of **{player_name}**" \
+                     f"({p_stars}) in {other_league} League {team_caviat}\n\n"
+          for player in other_players:
+              o_player_name, team_name = player[1], player[4]
+              stars = round((player[2] * 100)) / 100
+              rating_star = player[5] * 5
+              rating = round((rating_star * 100)) / 100
+              response += f"**{o_player_name}**: {stars} ({team_name}) - {rating}⭐ {raw_rating}\n"
+          return await ctx.send(response)
```

Comment:
discord message character limit is 2000 characters, if user indicates a large limit or less than 0 limit (unlimited) this response will fail with no indication to user what went wrong.

we *can* page the response (i think there's a helper method somewhere already to do this) but when used in busy public channels it gets very noisy hence the original limit of 10.